### ISSUE TITLE
Document retrieval for IDs matching `@base`

### ISSUE THREAD

**Issue by timathom** [terminusdb/terminusdb#1986]
**Describe the bug**
In TerminusCMS, I'm no longer able to retrieve documents whose IDs correspond to the `@base` IRI declared in my schema. Using the Python API, `client.get_document("person/f51cf591-1ddb-4c56-ba93-1200db8d4edb")` throws an error:
```
raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
```
But `client.get_document("lc:names/n2007180387")`, for a document whose ID matches a prefix (`lc:`) defined in my schema, does work.

If I build the latest version (11.1.4) from source, then I _am_ able to retrieve the document using the CLI:

```
./terminusdb doc get lux_sandbox/lux_collections_yale_edu_Sandbox/local/branch/main \
-l -g instance --unfold=false \
-i person/f51cf591-1ddb-4c56-ba93-1200db8d4edb > out.json
```

**To Reproduce**
1. Load [this schema](https://github.com/timathom/linked-art-terminusdb/blob/main/schema/CombinedLinkedArtSchemaExpanded.json) and [this data](https://github.com/timathom/linked-art-terminusdb/tree/main/test/dataset) into TerminusCMS.
2. Attempt to retrieve the document with ID `person/f51cf591-1ddb-4c56-ba93-1200db8d4edb` using the Python API, etc.

**Expected behavior**
All documents with IDs in the dataset should be retrievable.

**Info (please complete the following information):**
 - Latest TerminusCMS

**Comment by timathom** [comment]
In DFRNT, the following error is thrown when trying to fetch the doc with ID `person/f51cf591-1ddb-4c56-ba93-1200db8d4edb`:
```
Invalid document returned from TerminusDB, missing @id:
null
```

**Comment by hoijnet** [comment]
@timathom, correct. I sent a grab of the returned structure to @GavinMendelGleason for inspection as it looked odd to me.

The null comes from the inability to decipher a correct @id in the returned data.

**Comment by matko** [comment]
I cannot reproduce this.
Using the python client, I created a database in  TerminusCMS, loaded the schema and instance data you provided, and then retrieved the document with the command you specified (`client.get_document("lc:names/n2007180387")`). This returns the document.

So either we fixed this issue accidentally, or the actual issue is something else.
Are you able to reproduce with a fresh database? Or does this only occur in your working database?

**Comment by timathom** [comment]
@matko Indeed, I can retrieve with `client.get_document("lc:names/n2007180387")` but _not_ with `client.get_document("person/f51cf591-1ddb-4c56-ba93-1200db8d4edb")`, which was the command referenced above.

**Comment by timathom** [comment]
@matko, thanks--that makes sense! @hoijnet, is this something you can accommodate in DFRNT?

**Comment by timathom** [comment]
@matko I notice that with `unfold=false` subdocuments are also not expanded. Example:
```
{
  "@id": "digital/003e5c9f-2f78-49b7-b9c0-5f5ace8e4cdb",
  "@type": "dig:D1_Digital_Object",
  "crm:P129i_is_subject_of": [
    "Blank_Node/KFaAAyt9prA_sppF1",
    "Blank_Node/KFaAAyt9prA_sppF10",
    "Blank_Node/KFaAAyt9prA_sppF4",
    "Blank_Node/KFaAAyt9prA_sppF7"
  ],
  "crm:P1_is_identified_by": [
    "Blank_Node/KFaAAyt9prA_sppF13",
    "Blank_Node/KFaAAyt9prA_sppF14",
    "Blank_Node/KFaAAyt9prA_sppF16",
    "Blank_Node/KFaAAyt9prA_sppF18",
    "Blank_Node/KFaAAyt9prA_sppF19",
    "Blank_Node/KFaAAyt9prA_sppF20"
  ],
  "crm:P2_has_type": [
    "concept/77aec3ac-997b-42c2-91ba-0ab1669e3972",
    "concept/b34ad31c-4a2f-4cef-aee3-ae8d824e42a2"
  ],
  "crm:P67i_is_referred_to_by": [
    "Blank_Node/KFaAAyt9prA_sppF21",
    "Blank_Node/KFaAAyt9prA_sppF22",
    "Blank_Node/KFaAAyt9prA_sppF23",
    "Blank_Node/KFaAAyt9prA_sppF24"
  ],
  "rdfs:label": ["Trumbull ave [electronic resource]"],
  "la:digitally_carries": ["text/a33fb0b2-28e9-4825-af7f-e9739620addf"]
}
```
Here, the blank node objects of `crm:P129i_is_subject_of` are instances of an  `@unfoldable` class (`crm:E33_LinguisticObject`) in my schema, but the objects of `crm:P1_is_identified_by` are instances of a `@subdocument` class (`crm:E42_Identifier`).

I would still expect subdocuments to be expanded even with `unfold=false`. Ultimately, there are broader design issues involved, as discussed in https://github.com/orgs/terminusdb/discussions/1978.

**Comment by hoijnet** [comment]
> @matko, thanks--that makes sense! @hoijnet, is this something you can accommodate in DFRNT?

In most cases I prefer to use native functionality for these things, this is a corner care where I will need to think about and handle properly. I think we might want to figure out a mechanism in the Javascript client for resolving these properly. It could be that the unfoldable documents should be returned as an array of IDs somehow if they can't be unfolded, or there is a specific way that I can do the expansion in a generic way using a custom API.

Making it client-side feels non-trivial. I'll need to take a deeper look at it. Is is for display in the UI, custom REST API that it would make sense?

I think the round-trip of saving an assembled unfolded document again should work though saving it, right @matko? So it's only on retrieval we see the error?

Would it be possible to recognize the error on the client side and use an alternate flag to retrieve the base document and get the list of @unfoldable documents as a separate list? That way it would be "just" to fetch the extra list of documents, traverse the original one and replace if it's just for representation I the client. Server-side API is a bit more workpotentially due to the contextualization needed for the API and perhaps schema awareness, might be able to do without.
